### ISSUE TITLE
🐞  Python 3.9.7 causes tests to fail

### ISSUE THREAD

**Issue by keithrfung** [microsoft/electionguard-python#440]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Python 3.9.7 was found by the CI to cause issues with the existing test suite. 

### Expected Behavior

Tests should run as 3.9.5 and successfully pass. 

### Steps To Reproduce

1. Set environment to 3.9.7
This can be done with `poetry env use 3.9.7` and `pyenv local 3.9.7`

2. Run `make tests`

### Environment

```markdown
- OS: Mac, Windows, Linux
```

### Anything else?

This workflow shows the bugs if you wish to see the problems before starting
https://github.com/microsoft/electionguard-python/runs/3655735644?check_suite_focus=true

**Comment by ronnie-llamado** [comment]
I think the issue originates from [here](https://bugs.python.org/issue45081) and was recently [patched](https://github.com/python/cpython/pull/28121).

`electionguard.hash.CryptoHashable` and `electionguard.hash.CryptoHashCheckable` inherit `typing.Protocol`, so anything that then inherits from either of these two classes will show errors.

**Comment by keithrfung** [comment]
@ronnie-llamado Thanks for the information. Will have to test this!

**Comment by keithrfung** [comment]
@ronnie-llamado Did you happen to have any success getting this working with 3.9.7?

**Comment by ronnie-llamado** [comment]
I haven't pursued the issue further yet, but I can get on it. From some more looking around, there seems like two options:

1. Wait for the patch in the python source to be released in 3.9.8

2. Explicitly defined  `__init__` in dataclasses that inherit from `Protocol` (or in `electionguard`'s case from `CryptoHashable` or `CryptoHashCheckable`). Here's another project that took this path to resolve the same issue: [link](https://github.com/dbt-labs/dbt/pull/3855).

Here's the list of all dataclasses that inherit from `CryptoHashable` (no dataclasses inherit `CryptoHashCheckable`):
```python
src/electionguard/manifest.py:92:class AnnotatedString(CryptoHashable):
src/electionguard/manifest.py:111:class Language(CryptoHashable):
src/electionguard/manifest.py:130:class InternationalizedText(CryptoHashable):
src/electionguard/manifest.py:148:class ContactInformation(CryptoHashable):
src/electionguard/manifest.py:169:class GeopoliticalUnit(ElectionObjectBase, CryptoHashable):
src/electionguard/manifest.py:192:class BallotStyle(ElectionObjectBase, CryptoHashable):
src/electionguard/manifest.py:213:class Party(ElectionObjectBase, CryptoHashable):
src/electionguard/manifest.py:246:class Candidate(ElectionObjectBase, CryptoHashable):
src/electionguard/manifest.py:278:class SelectionDescription(OrderedObjectBase, CryptoHashable):
src/electionguard/manifest.py:305:class ContestDescription(OrderedObjectBase, CryptoHashable):
src/electionguard/manifest.py:522:class Manifest(CryptoHashable):
```
